### PR TITLE
Create how we use PRs page

### PR DESCRIPTION
**Pull request by U8NWXD**

### CODE REVIEW

File: How-we-use-pull-requests.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  Ready for review: PR awaiting review
+    Request changes: Changes requested
+      Addresses comments: (PR awaiting review)
+    All reviewers approve: PR labeled LGTM
+      Merge: PR closed
+```
+</details>
+
+1. (optional) Draft PR: Authors can open draft PRs to get early feedback or debug CI tests. Reviews at this stage are at the discretion of the author.
+2. PR ready for review: Once authors mark PRs as ready for review (or if they skip the draft PR stage entirely), reviews will automatically be requested from all code owners. Oppiabot will assign one developer (typically the developer from the changelog label) to take a first pass. Oppiabot will assign the other code owners to review once the first pass is complete.
```
Comment (DubeySandeep) [inline How-we-use-pull-requests.md:33]: The PR is expected to be reviewed by the main/primary reviewer first (Currently decided by OppiaBot) and once they approve, all other code owners get assigned!

You can find more details [here](https://docs.google.com/document/d/1y2_3Ns0V00QESZy3gswc413Rki-m1exTNDt3uVcu7w0/edit#).

File: How-we-use-pull-requests.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+5. PR merged: PRs can be merged once CI checks pass and the LGTM label has been applied. Only members of the Oppia organization can merge PRs.
+
+## Draft pull requests
+
+While making a contribution, you may discover that your change is not complete and needs some more work. You can make a draft pull request to make it easy for other developers to give you feedback; however, draft pull requests consume resources like time on our CI runners, which slows down CI runs for other developers. Hence, you should prefix the commit messages with `[skip ci]` or `[ci skip]` to disable CI runs when those commits are going to a draft PR. If you want those tests to run to help you debug a problem, you should open a PR against your fork's `develop` branch instead of the `oppia/oppia` develop. This will still delay other developers' Circle CI runs, but not their GitHub Actions runs.
+
+Learn more about skipping a [Travis CI build](https://docs.travis-ci.com/user/customizing-the-build/#skipping-a-build) and skipping a [Circle CI build](https://circleci.com/docs/2.0/skip-build/#skipping-a-build).
+
+## Labeling issues and pull requests
```
Comment (DubeySandeep) [inline How-we-use-pull-requests.md:44]: Let's not talk about issues here*

File: How-we-use-pull-requests.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
+## Table of contents
+
+* [PR lifecycle](#pr-lifecycle)
```
Comment (DubeySandeep) [inline How-we-use-pull-requests.md:3]: Not sure about the title of this page, this dock doesn't talk about the uses of the PRs, right?

File: How-we-use-pull-requests.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+4. LGTM: A PR gets labeled LGTM ("looks good to me") when it has sufficient reviews. This means that a code owner from each modified file has approved. When multiple developers share code ownership, only one has to approve.
+5. PR merged: PRs can be merged once CI checks pass and the LGTM label has been applied. Only members of the Oppia organization can merge PRs.
+
+## Draft pull requests
+
+While making a contribution, you may discover that your change is not complete and needs some more work. You can make a draft pull request to make it easy for other developers to give you feedback; however, draft pull requests consume resources like time on our CI runners, which slows down CI runs for other developers. Hence, you should prefix the commit messages with `[skip ci]` or `[ci skip]` to disable CI runs when those commits are going to a draft PR. If you want those tests to run to help you debug a problem, you should open a PR against your fork's `develop` branch instead of the `oppia/oppia` develop. This will still delay other developers' Circle CI runs, but not their GitHub Actions runs.
```
Comment (DubeySandeep) [inline How-we-use-pull-requests.md:40]: Can you please check with the OppiaBot team on what are the bot's expectations from draft PRs? (I'm unable recall all the expectations!)

File: How-we-use-pull-requests.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+All other labels are to be used on issues.
+
+It should be noted that the **good first issue** label should only be added by members of the onboarding team.
+
+A complete list of labels can be found [here](https://github.com/oppia/oppia/labels).
+
+## Juggling multiple pull requests
+
+After you've submitted a PR, and are waiting for it to get reviewed, you might want to start working on a new issue in the meantime. Git allows you to do this seamlessly! Always prioritize following up on your existing PRs and getting them ready for the next round of review. In particular, only start working on new issues/PRs after **all** the other PRs you're working on are in the "waiting for review" stage. This helps avoids creating a logjam in the build queues.
```
Comment (DubeySandeep) [inline How-we-use-pull-requests.md:68]: I don't see anything related to "Juggling" here, are we expecting to explain "how-tos" here or just inform that contributors can create multiple PRs in parallel? (If the latter one then maybe change the heading?)

File: How-we-use-pull-requests.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
+## Table of contents
+
+* [PR lifecycle](#pr-lifecycle)
+* [Draft pull requests](#draft-pull-requests)
+* [Labeling issues and pull requests](#labeling-issues-and-pull-requests)
+* [Juggling multiple pull requests](#juggling-multiple-pull-requests)
+
+## PR lifecycle
```
Comment (DubeySandeep) [inline How-we-use-pull-requests.md:8]: Should we call it a Pull request lifecycle? (Considering all other headings have "pull request")

**Review by DubeySandeep — CHANGES_REQUESTED** [review]
@U8NWXD I've left a few comments, PTAL!

File: How-we-use-pull-requests.md (hunk 1 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md:33]: Isn't that what's written? Specifically these sentences:

> Oppiabot will assign one developer (typically the developer from the changelog label) to take a first pass. Oppiabot will assign the other code owners to review once the first pass is complete.

File: How-we-use-pull-requests.md (hunk 4 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md:40]: Based on the recent draft PRs I've opened, I'm pretty sure that Oppiabot just expects to see `[skip ci]` or `[ci skip]` in the commit message.

@vojtechjelinek is this correct?

File: How-we-use-pull-requests.md (hunk 2 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md:44]: done

File: How-we-use-pull-requests.md (hunk 3 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md]: I've renamed it to "Pull requests at Oppia"

File: How-we-use-pull-requests.md (hunk 5 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md]: This is really just about handling multiple PRs in parallel. I've updated the heading

File: How-we-use-pull-requests.md (hunk 6 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md]: done

**Comment by U8NWXD** [comment]
@DubeySandeep I've addressed your comments PTAL

File: How-we-use-pull-requests.md (hunk 4 above)
Comment (U8NWXD) [reply How-we-use-pull-requests.md]: @vojtechjelinek has confirmed offline that Oppiabot just expects to see `[skip ci]` or `[ci skip]`

File: How-we-use-pull-requests.md (hunk 1 above)
Comment (DubeySandeep) [reply How-we-use-pull-requests.md]: > reviews will automatically be requested from all code owners

Ah, I misinterpreted "requested" as assigned! (Sorry!)

> You can find more details here.

Just to confirm, this wiki page follows the process defined in the doc linked above, right?

File: Pull-requests-at-Oppia.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
+## Table of contents
+
+* [PR lifecycle](#pr-lifecycle)
+* [Draft pull requests](#draft-pull-requests)
+* [Labeling issues and pull requests](#labeling-issues-and-pull-requests)
+* [Juggling multiple pull requests](#juggling-multiple-pull-requests)
```
Comment (DubeySandeep) [inline Pull-requests-at-Oppia.md:6]: Do we need to change the id name here to match the updated title?

**Review by DubeySandeep — APPROVED** [review]
LGTM, @seanlip Can you please take a final pass?

File: Pull-requests-at-Oppia.md (hunk 7 above)
Comment (DubeySandeep) [reply Pull-requests-at-Oppia.md:6]: Also, the heading names are not updated here?

File: Pull-requests-at-Oppia.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+Draft PR
+  Ready for review: PR awaiting review
+    Request changes: Changes requested
+      Addresses comments: (PR awaiting review)
+    All reviewers approve: PR labeled LGTM
+      Merge: PR closed
+```
+</details>
+
+1. (optional) Draft PR: Authors can open draft PRs to get early feedback or debug CI tests. Reviews at this stage are at the discretion of the author.
```
Comment (seanlip) [inline Pull-requests-at-Oppia.md:32]: Perhaps elaborate more on what "Reviews at this stage are at the discretion of the author" means. In particular what is the author supposed to do / what decision are they supposed to make here, and how do they communicate this decision?

File: Pull-requests-at-Oppia.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+4. LGTM: A PR gets labeled LGTM ("looks good to me") when it has sufficient reviews. This means that a code owner from each modified file has approved. When multiple developers share code ownership, only one has to approve.
+5. PR merged: PRs can be merged once CI checks pass and the LGTM label has been applied. Only members of the Oppia organization can merge PRs.
+
+## Draft pull requests
+
+While making a contribution, you may discover that your change is not complete and needs some more work. You can make a draft pull request to make it easy for other developers to give you feedback; however, draft pull requests consume resources like time on our CI runners, which slows down CI runs for other developers. Hence, you should prefix the commit messages with `[skip ci]` or `[ci skip]` to disable CI runs when those commits are going to a draft PR. If you want those tests to run to help you debug a problem, you should open a PR against your fork's `develop` branch instead of the `oppia/oppia` develop. This will still delay other developers' Circle CI runs, but not their GitHub Actions runs.
```
Comment (seanlip) [inline Pull-requests-at-Oppia.md:40]: Here and elsewhere, CircleCI is one word.

File: Pull-requests-at-Oppia.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+Learn more about skipping a [Travis CI build](https://docs.travis-ci.com/user/customizing-the-build/#skipping-a-build) and skipping a [Circle CI build](https://circleci.com/docs/2.0/skip-build/#skipping-a-build).
```
Comment (seanlip) [inline Pull-requests-at-Oppia.md:42]: Why do we still refer to Travis?

File: Pull-requests-at-Oppia.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+* `REVIEWERS: Please add changelog label`: When a new contributor opens a PR, they don't have permission to add labels. In that case, Oppiabot adds this label to tell reviewers to assign the appropriate changelog label.
+* `stale`: This label gets added by oppiabot to PRs that have been inactive for a week. They will be automatically closed after 4 more days of inactivity.
+* `PR: LGTM`: Indicates that a PR has all necessary approvals. The PR can be merged as soon as the CI checks pass.
+* Other labels starting with `PR`.
+
+A complete list of labels can be found [here](https://github.com/oppia/oppia/labels).
+
+## Working on multiple pull requests
+
+After you've submitted a PR, and are waiting for it to get reviewed, you might want to start working on a new issue in the meantime. Git allows you to do this seamlessly! Always prioritize following up on your existing PRs and getting them ready for the next round of review. In particular, only start working on new issues/PRs after **all** the other PRs you're working on are in the "waiting for review" stage. This helps avoids creating a logjam in the build queues.
```
Comment (seanlip) [inline Pull-requests-at-Oppia.md:64]: The "Git allows you to do this seamlessly!" part seems a bit of a throwaway. Either elaborate on it or link it to a separate page -- it's a reasonable "new contributor confusion" that they aren't sure how to make a parallel PR and we should try to explain it somehow.

The next point is a separate point and should be given its own prominence (perhaps as a separate paragraph).

File: Pull-requests-at-Oppia.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+5. PR merged: PRs can be merged once CI checks pass and the LGTM label has been applied. Only members of the Oppia organization can merge PRs.
+
+## Draft pull requests
+
+While making a contribution, you may discover that your change is not complete and needs some more work. You can make a draft pull request to make it easy for other developers to give you feedback; however, draft pull requests consume resources like time on our CI runners, which slows down CI runs for other developers. Hence, you should prefix the commit messages with `[skip ci]` or `[ci skip]` to disable CI runs when those commits are going to a draft PR. If you want those tests to run to help you debug a problem, you should open a PR against your fork's `develop` branch instead of the `oppia/oppia` develop. This will still delay other developers' Circle CI runs, but not their GitHub Actions runs.
+
+Learn more about skipping a [Travis CI build](https://docs.travis-ci.com/user/customizing-the-build/#skipping-a-build) and skipping a [Circle CI build](https://circleci.com/docs/2.0/skip-build/#skipping-a-build).
+
+## Labeling pull requests
```
Comment (seanlip) [inline Pull-requests-at-Oppia.md:44]: This list is kind of overwhelming. Some of this stuff is automatically added and some is meant to be added by the reviewer.

Perhaps explain:
- this is the stuff you should add (then have a bullet list saying "if your PR has X, add this label"); 
- this is the stuff that may get automatically added and what it means (this bullet list can be based on label);
- here's what you do if you don't have permissions to add labels.

**Review by seanlip — CHANGES_REQUESTED** [review]

File: Pull-requests-at-Oppia.md (hunk 12 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md:44]: done

File: Pull-requests-at-Oppia.md (hunk 11 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

File: Pull-requests-at-Oppia.md (hunk 10 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: I removed Travis and added GitHub Actions

File: Pull-requests-at-Oppia.md (hunk 9 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

File: Pull-requests-at-Oppia.md (hunk 8 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

File: Pull-requests-at-Oppia.md (hunk 7 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: fixed

**Comment by U8NWXD** [comment]
@seanlip PTAL

File: Pull-requests-at-Oppia.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+
+Learn more about [skipping a CircleCI build](https://circleci.com/docs/2.0/skip-build/#skipping-a-build) and [skipping a GitHub Actions run](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/).
+
+## Labeling pull requests
+
+While contributing to Oppia, you will need to add different labels to issues or pull requests which you are working on. However, not all labels are allowed on issues and pull requests. Below are labels which can be applied to pull requests:
+
+* Added by PR author:
+
+  * `dependencies`: Should be added to pull requests that updates one or more dependencies.
```
Comment (seanlip) [inline Pull-requests-at-Oppia.md:50]: updates --> update

**Review by seanlip — CHANGES_REQUESTED** [review]
Thanks! This should be my last round of comments.

File: Pull-requests-at-Oppia.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+  * `PR: Affects datastore layer`: Indicates that a PR changes the datastore layer. Adding this label notifies the developers in charge of datastore stability so they can review the PR. This can also be added automatically.
+  * Changelog (labels containing `PR CHANGELOG`): Each PR should have one of these labels added to indicate what project this PR applies to. The developer mentioned in the label will be assigned to review the PR first.
+* `PR: require post-merge sync to HEAD`: Should only be applied to pull requests which when merged will require all other open pull requests to be updated with the develop branch.
+
+* Added automatically:
+
+  * Labels starting with `PR: don't merge`: Indicate that some problem with the PR needs to be addressed before merging. For example, one of these labels gets added if the PR author hasn't signed the CLA yet.
```
[the remark this answers is not in the document]
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

File: Pull-requests-at-Oppia.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+## Labeling pull requests
+
+While contributing to Oppia, you will need to add different labels to issues or pull requests which you are working on. However, not all labels are allowed on issues and pull requests. Below are labels which can be applied to pull requests:
+
+* Added by PR author:
+
+  * `dependencies`: Should be added to pull requests that updates one or more dependencies.
+  * `PR: Affects datastore layer`: Indicates that a PR changes the datastore layer. Adding this label notifies the developers in charge of datastore stability so they can review the PR. This can also be added automatically.
+  * Changelog (labels containing `PR CHANGELOG`): Each PR should have one of these labels added to indicate what project this PR applies to. The developer mentioned in the label will be assigned to review the PR first.
+* `PR: require post-merge sync to HEAD`: Should only be applied to pull requests which when merged will require all other open pull requests to be updated with the develop branch.
```
[the remark this answers is not in the document]
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

File: Pull-requests-at-Oppia.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+Learn more about [skipping a CircleCI build](https://circleci.com/docs/2.0/skip-build/#skipping-a-build) and [skipping a GitHub Actions run](https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/).
+
+## Labeling pull requests
+
+While contributing to Oppia, you will need to add different labels to issues or pull requests which you are working on. However, not all labels are allowed on issues and pull requests. Below are labels which can be applied to pull requests:
+
+* Added by PR author:
+
+  * `dependencies`: Should be added to pull requests that updates one or more dependencies.
+  * `PR: Affects datastore layer`: Indicates that a PR changes the datastore layer. Adding this label notifies the developers in charge of datastore stability so they can review the PR. This can also be added automatically.
```
[the remark this answers is not in the document]
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

File: Pull-requests-at-Oppia.md (hunk 13 above)
Comment (U8NWXD) [reply Pull-requests-at-Oppia.md]: done

**Comment by U8NWXD** [comment]
@seanlip PTAL

**Review by seanlip — APPROVED** [review]
Thanks -- LGTM

**Pull request merged**